packages/react-data-grid/src/masks/InteractionMasks.js

```diff
@@ -296,23 +315,60 @@ class InteractionMasks extends React.Component {
 
   changeCellFromEvent(e) {
     e.preventDefault();
+    const isTab = e.keyCode === keyCodes.Tab;
+    const isShift = e.shiftKey;
+
+    if (isTab) {
+      this.changeCellFromTabAction(e);
+    } else if (isShift) {
+      this.changeSelectedRangeFromArrowKeyAction(e);
+    } else {
+      this.changeCellFromArrowKeyAction(e);
+    }
+  }
+
+  changeCellFromArrowKeyAction(e) {
+    const currentPosition = this.state.selectedPosition;
+    const { cellNavigationMode } = this.props;
+    const next = this.getNextPositionAndFireOnHitBoundaryIfNeccesary(e, currentPosition, cellNavigationMode);
+
+    this.selectCell(next);
+  }
+
+  changeCellFromTabAction(e) {
```

Comment:
changeCellFromArrowKeyAction and changeCellFromTabAction are practically the same function. It should be consolidated to changeCellFromArrowKeyAction, passing in the relevant cellNavigationMode as a parametre